### ISSUE TITLE
[Bug]: 5.4.2 fails to build on 32bit

### ISSUE THREAD

**Issue by eworm-de** [tukaani-project/xz#44]
### Describe the bug

Building the multilib package for Arch Linux (https://archlinux.org/packages/multilib/x86_64/lib32-xz/) fails. The lines producing the error did not change recently, but the version before (5.4.1) built without problems (and still builds with current toolchain).

### Version

latest

### Operating System

Arch Linux

### Relevant log output

```shell
In file included from lz/lz_encoder_mf.c:16:
../../src/liblzma/common/memcmplen.h: In function 'lzma_memcmplen':
../../src/liblzma/common/memcmplen.h:92:36: error: conversion to 'uint32_t' {aka 'const unsigned int'} from 'int' may change the sign of the result [-Werror=sign-conversion]
   92 |                 const uint32_t x = 0xFFFF ^ _mm_movemask_epi8(_mm_cmpeq_epi8(
      |                                    ^~~~~~
```

**Comment by Larhzu** [comment]
Sorry. :-( I added a few warning flags since I thought I had silenced them all.

Arch uses --enable-werror so that's why warnings make the build fail. This is good for testing :-) although it can cause annoyances like this. It was only recently that -Wno-format-truncation could be removed from the PKGBUILD file in Arch.

It's fixed in master and v5.4 now. Thanks!

**Comment by eworm-de** [comment]
Thanks a lot for the quick fix!
